src/DebuggerData.cpp

```diff
@@ -321,18 +332,57 @@ int Breakpoints::findBreakpoint(quint16 addr)
 			   }
    			);
 	if (i != breakpoints.end() && i->address == addr) {
-		return i - breakpoints.begin();
+		currentIndex = i - breakpoints.begin();
+	} else {
+		currentIndex = -1;
 	}
-	return -1;
+	return currentIndex;
 }
 
 int Breakpoints::findNextBreakpoint()
 {
-	// stub
-	// will implement findfirst/findnext scheme for speed
+	if (currentIndex < breakpointCount() - 1) {
+		return ++currentIndex;
+	}
 	return -1;
 }
 
+QStringList AttributeNames = {
+	"type",
+	"id",
+	"address",
+	"regionEnd",
+	"ps",
+	"ss",
+	"segment",
+	"condition",
+};
+
+QVariant Breakpoints::getAttribute(int index, QString attrName) const
+{
+	switch (AttributeNames.indexOf(attrName))
+	{
+		case 0:
+			return breakpoints[index].type;
+		case 1:
+			return breakpoints[index].id;
+		case 2:
+			return breakpoints[index].address;
+		case 3:
+			return breakpoints[index].regionEnd;
+		case 4:
+			return breakpoints[index].ps;
+		case 5:
+			return breakpoints[index].ss;
+		case 6:
+			return breakpoints[index].segment;
+		case 7:
+			return breakpoints[index].condition;
+		default:
+			break;
+	}
+	return QVariant();
+}
```

Comment:
Hmm, I looked at all callers of this method. I think the `attrName` parameter is always a fixed string, right?

Why pass `attrName` as a string? Can't it be an enum, then you can use that enum-value directly as the switch-value, and then you don't need the `AttributeNames` list.

Or going one step further, why not remove that parameter and instead have methods like `getAttributeType()`, `getAttributeId()`, ..., `getAttributeCondition()`? Extra advantage is that you can use the actual return type instead of a `QVariant`. Actually, `getAddress()` is already an example of this.

Or alternatively you can make a single
```const BreakPoint& getBreakpoint(int index) const;```
method. Then callers can do something like `getBreakPoint(id).address`. This requires exposing `struct BreakPoint` (=move it to the public section of `Breakpoints`). But you effectively already do that via the current `getAttribute()` method. I mean you already expose the same information, only in a more obscure way.

Reply:
This is fixed now. I didn't want to put much time into something that I know I would probably have to change again. So I was waiting for you to tell me. 😅